### ISSUE TITLE
Limit to groups

### ISSUE THREAD

**Issue by GMo7944** [nextcloud/maps#170]
Hello
I want that only some users can use maps.
Then I tried to use the "Limit to groups" fonctionnality.

But it disappeared. I'm pretty sure I saw it at the firt installation but now impossible to see it.
I uninstall, reinstall => nothing.

Is there a trick ? 

I'm running the 16.0.4

Also : thank you for your work.

**Comment by eneiluj** [comment]
Well I can't see "limit to groups" for Maps app neither. I don't know why :smile: because it appears for my other NC apps.

@juliushaertl @nickvergessen @rullzer Do you have a clue about a potential reason why Maps app can't be limited to groups?

**Comment by nickvergessen** [comment]
https://github.com/nextcloud/maps/blob/master/appinfo/info.xml#L57

It is due to the type filesystem. I think that is wrong in general and we need a different solution. But i guess this is because the app needs to be loaded on dav requests, so it can index the locations of images and gson files?

**Comment by eneiluj** [comment]
@nickvergessen Thanks for your answer. Yep, the type filesystem is needed to make file hooks work (dav or not).

> I think that is wrong in general and we need a different solution.

I guess you somehow mean there must be a way to express that an app manipulates the filesystem but still be able to make group limitations and other stuff.

Well @GMo7944 it seems group limitations are not going to be possible soon. Sorry.

**Comment by nickvergessen** [comment]
> I guess you somehow mean there must be a way to express that an app manipulates the filesystem but still be able to make group limitations and other stuff.

Yes. We need to find a solution, so hooks are still/always triggered, but the UI is not accessible. I had the same problem with quite some other apps already

**Comment by szaimen** [comment]
I would also really like to see this feature :)

**Comment by Thatoo** [comment]
> 1. disabled maps
> 2. removed the following lines in maps/appinfo/info.xml
>   `     <types> <filesystem/> </types>`
> 3. enabled maps

@eneiluj do you know if that has any bad side effects?

_Originally posted by @szaimen in https://github.com/nextcloud/maps/issues/507#issuecomment-759107729_

**Comment by Thatoo** [comment]
This trick does work pn Nextcloud 25 with last version of Maps

**Comment by tacruc** [comment]
@Thatoo did you check, that the filesystem Hooks are still executed correctly?

**Comment by Thatoo** [comment]
No. How can I do that?

**Comment by nickvergessen** [comment]
> do you know if that has any bad side effects?

App might be skipped in webdav and therefor not get events to index new photos (as tacruc already points out).

**Comment by tacruc** [comment]
@nickvergessen do we know what this might is dependent on?

Is it either skipped or not skipped, or might it be skipped in some environments and not skipped in other environments?
Or even worse might it be skipped sometimes on the same env?

I'm just woundering if it is sufficent to test once if it is skipped, do we have to test it one on all NC version, or do we have to test once on all possible enviroments (db, chaching, operating system, arichtecture ...)

**Comment by tacruc** [comment]
Anyhow I think the best solution for this and many other images is to replace the photo backend, by a dav search for images with geolocation. (which should be possible in NC25 or NC26.

**Comment by nickvergessen** [comment]
> Is it either skipped or not skipped, or might it be skipped in some environments and not skipped in other environments?
Or even worse might it be skipped sometimes on the same env?

I would need to dive into more code I don't have in my brain atm to answer this. But we load some apps based on types, etc.

Also note that when maps is not enabled for user A, and they upload a photos to a folder shared from user B, the maps app would not be loaded for user A and therefor not follow the photo upload, although user B might expect that.
